zzibo iP

zzibo opened the pull request:
### Hinlok Pro

> "你给我这一辈子都不想失恋的爱" — Hinlok

**Hinlok Pro** enhances your productivity by organising tasks and streamlining workflows, with Hinlok he:

- **Is handsome**: Saves time with intuitive task management
- **Got aura**: Adapts to your workflow seamlessly
- **Can Sing**: Available on multiple platforms

1. **Simple Installation**:
   - Download the application.
   - Install with a few clicks.
   - Get started right away.

**Features include:**

- [ ] Managing tasks effectively
- [ ] Setting deadlines and reminders
- [ ] Integration with calendars (upcoming)

If you're a developer, here's how you can use it to manage tasks programmatically:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

zzibo committed:
renamed duke and added echo function

zzibo committed:
renamed duke and added echo function

zzibo committed:
added features to add todo and show todos

zzibo committed:
added features to add todo as Done and Undone

zzibo committed:
renamed and added Todo, Deadline and Event classes

zzibo committed:
added test case and tested all features

zzibo committed:
removed unnecessary toString()

zzibo committed:
added exception and exception handling

zzibo committed:
add delete function

zzibo committed:
add save and load feature

zzibo committed:
fix warnings

zzibo committed:
Merge branch 'branch-Level-7'

zzibo committed:
change by from string to LocalDate

zzibo committed:
add method to find deadline task using date

zzibo committed:
Merge branch 'branch-Level-8'

zzibo committed:
add more OOP

zzibo committed:
add more OOP classes

zzibo committed:
fix unused imports

zzibo committed:
package all classes under hinlok package

zzibo committed:
Merge branch 'add-gradle-support'

zzibo committed:
install gradle

zzibo committed:
add Junit test and create JAR file

zzibo committed:
Adjust code to Coding Standards

zzibo committed:
Add Java Docs

zzibo committed:
Add find command

zzibo committed:
Debug add command

zzibo committed:
Merge branch 'branch-A-CodingStandard'

zzibo committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/hinlok/command/Command.java
#	src/main/java/hinlok/file/TaskFile.java
#	src/main/java/hinlok/parser/Parser.java
#	src/main/java/hinlok/ui/Ui.java

zzibo committed:
Debug parser after merge

zzibo committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/hinlok/parser/Parser.java

zzibo committed:
Merge level-9, java-doc and coding standard branches

wrjgold commented on src/main/java/hinlok/Hinlok.java:
```diff
@@ -0,0 +1,42 @@
+import hinlok.tasks.TaskList;
+import hinlok.file.TaskFile;
+import hinlok.ui.Ui;
+
```
Do you think writing javadocs will make the class and methods' purpose more clear?

zzibo replied:
yes, will do!

wrjgold commented on src/main/java/hinlok/file/TaskFile.java:
```diff
@@ -0,0 +1,130 @@
+    /**
+     * Returns a ArrayList of Tasks stored in the file
+     * @return ArrayList of tasks
+     */
```
I like the use of documentation here!

wrjgold commented on src/main/java/hinlok/parser/Parser.java:
```diff
@@ -0,0 +1,73 @@
+package hinlok.parser;
+
+import hinlok.command.Command;
+import hinlok.command.ListCommand;
+import hinlok.command.ExitCommand;
+import hinlok.command.DeleteCommand;
+import hinlok.command.MarkCommand;
+import hinlok.command.UnmarkCommand;
+import hinlok.command.AddCommand;
+import hinlok.command.FindCommand;
```
I like how you didn't use a wildcard import so that it's clear what you're importing

zzibo replied:
thank you!

wrjgold commented on src/main/java/hinlok/ui/Ui.java:
```diff
@@ -0,0 +1,57 @@
+     *
+     * @param errorMessage
+     */
+
```
Consider removing blank line to be consistent with your other documentations

zzibo replied:
Will do !

wrjgold reviewed (commented):
I like your clean code! However do consider writing JavaDocs for all your public class and methods!

leowolf275 commented on src/main/java/hinlok/command/Command.java:
```diff
@@ -0,0 +1,16 @@
+import hinlok.tasks.TaskList;
+import hinlok.ui.Ui;
+
+public abstract class Command {
```
Great idea to separate each of the individual commands into their own separate classes!

leowolf275 commented on src/main/java/hinlok/parser/Parser.java:
```diff
@@ -0,0 +1,73 @@
+import hinlok.exceptions.HinlokException;
+
+public class Parser {
+    public enum CommandType {
```
Glad to see you did the optional task of using Enums! Good job!

leowolf275 commented on src/main/java/hinlok/Hinlok.java:
```diff
@@ -0,0 +1,42 @@
+        Hinlok hinlok = new Hinlok("./data/hinlok.txt");
+        hinlok.chat();
+    }
+
```
Seems to me that you are missing the Storage class specified in the instructions. Is there any reason you decided against implementing it?

leowolf275 commented on src/test/java/hinlok/HinlokTest.java:
```diff
@@ -0,0 +1,45 @@
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class HinlokTest {
+    @Test
```
Great tests! Consider separating each of the tests into different classes!

leowolf275 reviewed (commented):
Overall LGTM! As the previous reviewer said, pls don't forget to include JavaDoc throughout the presentation. Keep up the good work!

zzibo committed:
Implement checkstyle and JavaFX

zzibo committed:
Remove bugs in MainWindow

zzibo committed:
Add JavaDocs to Hinlok.java and Ui.java

zzibo committed:
Merge branch 'branch-Level-10'

zzibo committed:
Adjust files to implement javaFX

- remove Ui class as it has a lot of useless implementation
- remove text based ui
- improve abstraction for task list

zzibo committed:
Merge pull request #1 from zzibo/branch-A-Assertions

Branch a assertions

zzibo committed:
Improve SLAP and Happiest path in TaskFile.java

zzibo committed:
Merge pull request #2 from zzibo/branch-A-CodeQuality

Improve SLAP and Happiest path in TaskFile.java

zzibo committed:
Add new feature to see duplicated tasks

- Add new case in parser
- Add new DetectDuplicateCommand
- type duplicate to see duplicated tasks and their index

zzibo committed:
Remove errors

zzibo committed:
Implement new feature to detect and deal with duplicates

Duplicated task are present as there are no warning when
duplicated task are added

Let's add a detectDuplicate command and add a warning when users add in new task with the same type and name

Alert the user when a duplicated task is added instead of stopping the user from adding to give users more autonomy

zzibo committed:
MainWindow.java: Improve GUI of MainWindow

The current GUI needs to be improved to give better user experience

Changing the background, text bubble and profile picture will improve the user experience of this chatbot. Furthermore, delaying the response of chatbot makes the interaction more realistic

Using the chosen images and delaying the chatbot makes the chatbot more unique and provide a better experience

Code assisted by ChatGPT

zzibo committed:
Add hinlok tests and javadocs

Previous tests did not test the new detect duplicate feature

Testing is necessary to ensure it return the expected output

Let's add test that check the output of list and duplicate

zzibo committed:
Add Readme and include a user guide

This guide will provide users with instruction on how to interact with the chatbot

Let's edit the Readme file such that it explain the basic features and commands to use

zzibo committed:
Merge branch 'user-guide' into more-hinlok-tests

zzibo committed:
Merge pull request #4 from zzibo/more-hinlok-tests

Add more hinlok tests to test out duplicate and list features

Add a readme file that displays a user guide
